modules/concurrency/include/hephaestus/concurrency/any_sender.h

```diff
@@ -0,0 +1,291 @@
+  }
+  ~AnyOperation() = default;
+  AnyOperation(AnyOperation&& other) = delete;
+  auto operator=(AnyOperation&& other) -> AnyOperation& = delete;
+  AnyOperation(const AnyOperation& other) = delete;
+  auto operator=(const AnyOperation& other) -> AnyOperation& = delete;
+
+  void start() & noexcept {
+    impl_->start();
+  }
+
+private:
+  std::unique_ptr<Base> impl_;
+};
+
+template <typename T>
+struct AnySenderBase {
+  virtual ~AnySenderBase() = default;
+
+  virtual auto doConnect(internal::AnyReceiver<T> receiver) -> internal::AnyOperation = 0;
+};
+
+template <typename T, typename Sender>
+struct AnySenderImpl : AnySenderBase<T> {
+  static_assert(!std::is_const_v<Sender>);
+  static_assert(!std::is_reference_v<Sender>);
+  Sender sender;
+  template <typename SenderT>
+    requires(std::is_same_v<std::decay_t<SenderT>, Sender>)
+  explicit AnySenderImpl(SenderT&& outer_sender) : sender(std::forward<SenderT>(outer_sender)) {
+  }
+
+  auto doConnect(internal::AnyReceiver<T> receiver) -> internal::AnyOperation final {
+    return internal::AnyOperation{ std::move(sender), std::move(receiver) };
+  }
+};
+}  // namespace internal
+
+template <typename T>
+class AnySender;
+
+template <typename T>
+struct SelectValueCompletion {
+  using TypeT = stdexec::set_value_t(T);
+};
+template <>
+struct SelectValueCompletion<void> {
+  using TypeT = stdexec::set_value_t();
+};
+
+template <typename T>
+using SelectValueCompletionT = typename SelectValueCompletion<T>::TypeT;
+
+template <typename Sender, typename T>
+concept AnySenderRequirements =
+    !std::is_same_v<std::decay_t<Sender>, AnySender<T>> &&
+    (stdexec::sender_of<Sender, SelectValueCompletionT<T>, internal::AnyEnv> ||
+     stdexec::sender_of<Sender, stdexec::set_stopped_t(), internal::AnyEnv> ||
+     stdexec::sender_of<Sender, stdexec::set_error_t(std::exception_ptr), internal::AnyEnv>);
+
+/// Implementation for a type erased sender.
+///
+/// @tparam T The value this sender completes with
+template <typename T>
+class AnySender {
+public:
+  using sender_concept = stdexec::sender_t;
+  using completion_signatures =
+      stdexec::completion_signatures<SelectValueCompletionT<T>, stdexec::set_error_t(std::exception_ptr),
+                                     stdexec::set_stopped_t()>;
+
+  template <AnySenderRequirements<T> Sender, typename SenderT = std::decay_t<Sender>>
+  AnySender(Sender&& sender)  // NOLINT(google-explicit-constructor, hicpp-explicit-conversions)
+    : impl_(std::make_unique<internal::AnySenderImpl<T, SenderT>>(std::forward<Sender>(sender))) {
+  }
+
+  ~AnySender() = default;
+  AnySender(AnySender&& other) = default;
+  auto operator=(AnySender&& other) -> AnySender& = default;
+  AnySender(const AnySender& other) = delete;
+  auto operator=(const AnySender& other) -> AnySender& = delete;
+
+  template <stdexec::receiver_of<completion_signatures> Receiver>
+  auto connect(Receiver&& receiver) {
```

Comment:
```suggestion
  auto connect(Receiver&& receiver) && {
```

To avoid the sender being moved twice, right? Because after the type erasure, the sender is always moved in `doConnect`.